src/UI/Components/Flipcoin.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {useEffect, useRef} from 'react';
+import {Animated, SafeAreaView, StyleSheet} from 'react-native';
+
+const Flipcoin = () => {
+  const animatedValue = useRef(new Animated.Value(0)).current;
+
+  const startFlipAnimation = () => {
+    animatedValue.setValue(0); // Reset the animated value
+    Animated.loop(
+      Animated.sequence([
+        Animated.timing(animatedValue, {
+          toValue: 180,
+          duration: 1000, // Duration of 1 second to flip to 180 degrees
+          useNativeDriver: true,
+        }),
+        Animated.timing(animatedValue, {
+          toValue: 0,
+          duration: 1000, // Duration of 1 second to flip back to 0 degrees
+          useNativeDriver: true,
+        }),
+      ]),
+    ).start();
+  };
+
+  useEffect(() => {
+    startFlipAnimation();
+  }, []);
+
+  const setInterpolate = animatedValue.interpolate({
```

Comment:
single line comment for this value about what is it's use

Reply:
Done